MeesGame/Gameplay/GameObjects/AnimatedMovingGameObject.cs

```diff
@@ -23,13 +24,13 @@ public override void Update(GameTime gameTime)
             {
                 if (gameTime.TotalGameTime - lastAnimationTime >= timeBetweenAnimations)
                 {
-                    Sprite.SheetIndex = (Sprite.SheetIndex + 1) % Sprite.NumberSheetElements;
+                    Sprite.SheetIndex = restingSpriteIndex + (Sprite.SheetIndex + 1) % this.Sprite.NumberColumns;
```

Comment:
This code only works when restingSpriteIndex is a multiple of NumberColumns. For example, if restingSpriteIndex would be 5 and NumberColumns 4, SheetIndex would take on the following values:

1. SheetIndex = 5 + (0 + 1) % 4 = 6
2. SheetIndex = 5 + (6 + 1) % 4 = 8

Another problem with your implementation is that updating the restingSpriteIndex will not immediately  update the SheetIndex. In C# you can use a setter for things like this. 

Maybe something like this:
```C#
protected int SpriteRowIndex{
    get{
        return SpriteIndex/NumberColumns;
    }
    set{
       SpriteIndex = value*NumberColumns + (SpriteIndex%NumberColumns);
    }
}   
```

You could even put this in the `SpriteSheet`  class instead of the animation class.